apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -69,23 +62,42 @@ defmodule Aecore.Peers.Worker do
     GenServer.cast(__MODULE__, {:broadcast_to_all, {type, data}})
   end
 
+  @doc """
+  Gets a random peer nonce
+  """
+  @spec get_peer_nonce() :: integer()
+  def get_peer_nonce() do
+    case :ets.info(:nonce_table) do
+      :undefined -> create_nonce_table()
+      _ -> :table_created
+    end
+    case :ets.lookup(:nonce_table, :nonce) do
+      [] ->
+        rand = :rand.uniform(@mersenne_prime)
+        :ets.insert(:nonce_table, {:nonce, rand})
+        rand
+      _ ->
+        :ets.lookup(:nonce_table, :nonce)[:nonce]
+    end
+  end
+
   ## Server side
 
   def init(initial_peers) do
     {:ok, initial_peers}
   end
 
-  def handle_call({:add_peer,uri}, _from, %{peers: peers, nonce: own_nonce} = state) do
```

Comment:
I would leave the the peers map wrapped in a state map. In future we might want to store more data in state.

Reply:
@HristianHristov will you adjust this as well?